manifests/config.pp

```diff
@@ -16,24 +16,106 @@
+          cluster_name                        => $vault::cluster_name,
+          cache_size                          => $vault::cache_size,
+          disable_cache                       => $vault::disable_cache,
+          disable_mlock                       => $vault::disable_mlock,
+          plugin_directory                    => $vault::plugin_directory,
+          plugin_file_uid                     => $vault::plugin_file_uid,
+          plugin_file_permissions             => $vault::plugin_file_permissions,
+          telemetry                           => $vault::telemetry,
+          default_lease_ttl                   => $vault::default_lease_ttl,
+          max_lease_ttl                       => $vault::max_lease_ttl,
+          default_max_request_duration        => $vault::default_max_request_duration,
+          detect_deadlocks                    => $vault::detect_deadlocks,
+          raw_storage_endpoint                => $vault::raw_storage_endpoint,
+          introspection_endpoint              => $vault::introspection_endpoint,
+          ui                                  => $vault::enable_ui,
+          pid_file                            => $vault::pid_file,
+          enable_response_header_hostname     => $vault::enable_response_header_hostname,
+          enable_response_header_raft_node_id => $vault::enable_response_header_raft_node_id,
+          log_level                           => $vault::log_level,
+          log_format                          => $vault::log_format,
+          log_file                            => $vault::log_file,
+          log_rotate_duration                 => $vault::log_rotate_duration,
+          log_rotate_bytes                    => $vault::log_rotate_bytes,
+          log_rotate_max_files                => $vault::log_rotate_max_files,
+          experiments                         => $vault::experiments,
+          api_addr                            => $vault::api_addr,
+          cluster_addr                        => $vault::cluster_addr,
+          disable_clustering                  => $vault::disable_clustering,
+          disable_sealwrap                    => $vault::disable_sealwrap,
+          disable_performance_standby         => $vault::disable_performance_standby,
+          license_path                        => $vault::license_path,
+          replication                         => $vault::replication,
+          sentinel                            => $vault::sentinel,
+          service_registration                => $vault::service_registration,
+          log_requests_level                  => $vault::log_requests_level,
+          entropy_augmentation                => $vault::entropy_augmentation,
+          kms_library                         => $vault::kms_library,
+          extra_config                        => $vault::extra_config,
+        }
+      )
+    }
     file { "${vault::config_dir}/config.json":
```

Comment:
This file path is not really correct anymore, and furthermore there shouldn't be hcl in a file called .json.
Note that the package-provided service has forced the use of a different file for quite some time:
https://github.com/hashicorp/vault/blob/main/.release/linux/package/usr/lib/systemd/system/vault.service#L23

The location of this file should be another parameter, but if the user sets manage repo, changing this parameter would need to be blocked and forced to this path given by hashicorp.

Reply:
I think the path should be correct here. It is `/etc/vault.d` if the user is installing from a repo or `/etc/vault` if they aren't. If that isn't correct, what _should_ it be?